modules/core/src/test/java/org/locationtech/jts/io/WKBTest.java

```diff
@@ -151,7 +151,7 @@ private void runWKBTest(String wkt) throws IOException, ParseException
 
 	private void runWKBTestPackedCoordinate(String wkt) throws IOException, ParseException {
 		GeometryFactory geomFactory = new GeometryFactory(
-				new PackedCoordinateSequenceFactory(PackedCoordinateSequenceFactory.DOUBLE, 2));
+				new PackedCoordinateSequenceFactory(PackedCoordinateSequenceFactory.DOUBLE));
```

Comment:
What's the default dimension of the PackedCoordinateSequenceFactory?

Reply:
dimension 3, measures 0, you can see in the constructors.

Reply:
Then this is a change, no?  Here we are testing getting a PCSF which creates things of dimension 2, right?

Also, a WKBReader should be able to return values which have any 2, 3, and 4 dimensions...

Reply:
Oh here is the deal, the previous implementation of the factory was modal! So it held on to the init value `2` and would use it as a default for some of the create methods.

I removed this modal functionality since it introduced the possibility for error, especially with the static final factory implementations. Any thread could change the "default" messing up another thread.

Sot the above change is on purpose.

Reply:
+1